### PR TITLE
[master] Deserialization optimizations

### PR DESCRIPTION
**Pull request by vangheem**

### CODE REVIEW

File: guillotina/tests/test_api.py [hunk 1]
```diff
@@ -1226,11 +1226,7 @@ class TestSchema(Item):
         )
         assert status == 412
         assert resp["deserialization_errors"] == [
-            {
-                "errors": [{"field": "tags", "message": "Expected <class 'str'> but found <class 'int'>."}],
-                "field": "tags",
-                "message": "Wrong contained type",
-            }
+            {"message": "Expected <class 'str'> but found <class 'int'>.", "field": "tags"}
```
Comment (masipcat) [inline guillotina/tests/test_api.py]: Is there any way we can keep the nested error list?

Comment (vangheem) [reply guillotina/tests/test_api.py]: yes, I'll see what I can do!

Comment (masipcat) [reply guillotina/tests/test_api.py]: I think adding this again will do the fix https://github.com/plone/guillotina/pull/818/files#diff-b2a845fe39a48b7b8c8bbeb584780abaL85

Also in IDict deserializer

Comment (masipcat) [reply guillotina/tests/test_api.py]: And IObject as well

Comment (vangheem) [reply guillotina/tests/test_api.py]: I think my latest change fixed it if you could take a look.

Comment (masipcat) [reply guillotina/tests/test_api.py]: Yes, now i see the code I was referring to is here: https://github.com/plone/guillotina/pull/818/files#diff-b2a845fe39a48b7b8c8bbeb584780abaR57

Comment (masipcat) [reply guillotina/tests/test_api.py]: Thank you :)

**Pull request merged**